index.py

```diff
@@ -1,73 +1,393 @@
+                year += 1 
+
+    return skisesonger
+
+#oppgåve d) 
+#graferSkiføre(x,y) - plottar trend og faktisk skiføre kvart år 
+#Parameter:
+#   [list] x = år
+#   [list] y = dagar med skiføre
+#ingen returverdi
+def graferSkifore(skiforeDict):
+    x = [*skiforeDict]
+    y = [*skiforeDict.values()]
+    startAar = x[0]
+    sluttAar = x[-1]
+    trend = g(x,y) #oppgåve c), returnerer ei liste med stigningstal og konstantledd for trenden
+    print(f"Trenden er {trend[0]}x + {trend[1]}")
+    trendGraf = []
+    trendGraf.append(f(trend[0], trend[1], 0)) #a,b,x
+    trendGraf.append(f(trend[0], trend[1], len(x)))
+    plt.plot(x,y, label="Dagar med skiføre kvart år")
+    plt.plot([startAar, sluttAar], trendGraf, label="Trend")
+    plt.show()
+
+#lineær funkjon
+def f(a,b,x): return a*x+b 
+
+##utføring av kode
+datasett = lesFraFil("snoedybder_vaer_en_stasjon_dogn.csv")
+objektForSkifore = skiforeDager(datasett["dato"], datasett["snodybde"])
+#oppgåve d)
+graferSkifore(objektForSkifore)
+
+#Oppgave f, finner lengste sammenhengende periode med tørke per år og plotter det
+def lengsteTorkePeriodePerAr(aDatoListe, aNedborListe):
+    years = dict()
+    yearString = list()
+    year = list()
+    currentYear = aDatoListe[0].year
+    #print(year)
+
+    for i, dato in enumerate(aDatoListe):
+        if not dato.year == currentYear:
+            for c in yearString:
+                try:
+                    c = c.replace(",", ".")
+                    year.append(float(c))
+                except ValueError:
+                    pass
+            
+            if len(year) >= 300:
+                lengsteTorkePeriode = lengdeLengsteSekvens(year)
+                years[currentYear] = lengsteTorkePeriode
+
+            year.clear()
+            yearString.clear()
+            currentYear += 1
+
+        if dato.year == currentYear:
+            yearString.append(aNedborListe[i])
+
+    plt.plot(years.keys(), years.values(), marker = "o")
+    plt.xlabel("Årstall")
+    plt.ylabel("Lengste sekvens med tørkedager")
+    plt.show()
+
+    def sum_over_5(temperaturdata):
+    summen = 0
+    for temp in temperaturdata:
+        if temp > 5:
+            summen += temp - 5
+    return summen
+
+def beregn_gyldighet_og_årsvekst(årsdict):
+    gyldige_år_liste_e = []
+    vekst_resultater = []
+    for år, temperaturdata in årsdict.items():
+        # Filtrer ut ugyldige år
+        if len(temperaturdata) >= 300:
+            vekst = sum_over_5(temperaturdata)
+            vekst_resultater.append((år, vekst))
+            gyldige_år_liste_e.append(år)
+    print(vekst_resultater)
+    #print(temperaturdata)
+    #print(årsdict.items())
+
+
+    # Plot resultatene
+    år, vekst = zip(*vekst_resultater)
+    plt.bar(år, vekst)
+    plt.xlabel('År')
+    plt.ylabel('Vekst over 5 grader')
+    plt.title('Vekst for den tenkte planten hvert år')
+    plt.show()
+
+    return gyldige_år_liste_e
+
+def main_e():
+    årsdict = dict()
+
```

Comment:
Hilsen Tord igjen: her begynner oppgave h). Kaller main_e og main_h i bunn av mine oppgaver for å kjøre funksjonene mine.